tap_hubspot/__init__.py

```diff
@@ -830,26 +822,40 @@ def sync_tickets(STATE, ctx):
         # To handle records updated between start of the table sync and the end,
         # store the current sync start in the state and not move the bookmark past this value.
         sync_start_time = utils.now()
-        for row in gen_request_tickets(stream_id, url, params, 'results', "paging"):
-            # parsing the string formatted date to datetime object
+        for row in get_v3_records(stream_id, url, params, 'results', "paging"):
+            # Parsing the string formatted date to datetime object
             modified_time = utils.strptime_to_utc(row[bookmark_key])
 
             # Checking the bookmark value is present on the record and it
             # is greater than or equal to defined previous bookmark value
             if modified_time and modified_time >= bookmark_value:
-                # transforms the data and filters out the selected fields from the catalog
+                # Transforms the data and filters out the selected fields from the catalog
                 record = transformer.transform(lift_properties_and_versions(row), schema, mdata)
                 singer.write_record(stream_id, record, catalog.get(
                     'stream_alias'), time_extracted=utils.now())
-            if modified_time and modified_time >= max_bk_value:
-                max_bk_value = modified_time
+                if modified_time >= max_bk_value:
+                    max_bk_value = modified_time
```

Comment:
Any reason, now, we are not going check if the modified_time is None or not?

Reply:
We are already checking it in the outer `if` condition.